File: Exercise/Hard/02. Caesar Cipher Encryption/Code/Caesar Cipher Encryption.py
def caesar_cipher(text, shift):
    encrypted_text = []
    for char in text:
        if char.isalpha():
            base = ord('A') if char.isupper() else ord('a')
            encrypted_char = chr((ord(char) - base + shift) % 26 + base)
            encrypted_text.append(encrypted_char)
        else:
            encrypted_text.append(char)
    return ''.join(encrypted_text)

def main():
    text = input()
    shift = int(input())

    encrypted_text = caesar_cipher(text, shift)
    print(encrypted_text)

if __name__ == "__main__":
    main()

File: Exercise/Hard/02. Caesar Cipher Encryption/Code/Caesar Cipher Encryption.cs
using System;

class Program {
    static void Main() {
        string input = Console.ReadLine();

        int shift = int.Parse(Console.ReadLine());

        string encryptedText = CaesarCipher(input, shift);
        Console.WriteLine(encryptedText);
    }

    static string CaesarCipher(string input, int shift) {
        char[] chars = input.ToCharArray();

        for (int i = 0; i < chars.Length; i++) {
            if (char.IsLetter(chars[i])) {
                char baseChar = char.IsUpper(chars[i]) ? 'A' : 'a';
                chars[i] = (char)(((chars[i] + shift - baseChar) % 26) + baseChar);
            }
        }

        return new string(chars);
    }
}



